.changes/store-plugin-rework.md

```diff
@@ -0,0 +1,24 @@
+---
+"store": minor:breaking
+---
+
+**Breaking change**: Renamed `StoreCollection` to `StoreState`
```

Comment:
We can no longer make breaking changes, we need to revert this or keep it as an alias with a deprecation warning.

Reply:
Isn't the stability of plugins are still undetermined?